### PR TITLE
added percentiles arg to summary command, unit tests

### PR DESCRIPTION
**Pull request by mitzimorris**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary

Added argument to CmdStanMCMC `summary` method allowing user to specify percentiles to report.
Added unit tests, while doing, made `diagnose` unit test a little less brittle.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):  Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: cmdstanpy/stanfit.py [hunk 1]
```diff
@@ -499,12 +499,34 @@ def _assemble_sample(self) -> None:
...
+
+        :param percentiles: Ordered non-empty list of percentiles to report.
+            Must be integers from (1, 99), inclusive.
         """
+        percentiles_str = '--percentiles=5,50,95'
+        if percentiles is not None:
+            if len(percentiles) == 0:
+                raise ValueError(
+                    'invalid percentiles argument, must be ordered'
+                    ' non-empty list from (1, 99), inclusive.'
```
Comment (ahartikainen) [inline cmdstanpy/stanfit.py:516]: are integers only approved?

File: cmdstanpy/stanfit.py [hunk 2]
```diff
@@ -499,12 +499,34 @@ def _assemble_sample(self) -> None:
...
+            for pct in percentiles:
+                if pct > 99 or not pct > cur_pct:
+                    raise ValueError(
+                        'invalid percentiles spec, must be ordered'
+                        ' non-empty list from (1, 99), inclusive.'
+                    )
+                cur_pct = pct
+            percentiles_str = '='.join(
+                ['--percentiles', ','.join([str(x) for x in percentiles])]
+            )
```
Comment (ahartikainen) [inline cmdstanpy/stanfit.py:529]:
```suggestion
            percentiles_str = '='.join(
                '--percentiles', 
                ','.join(str(x) for x in percentiles)
            )
```

I think generator expr should work here

File: cmdstanpy/stanfit.py (hunk 1 above)
Comment (mitzimorris) [reply cmdstanpy/stanfit.py:516]: yes, stansummary only does integer percentiles - no 99.99%  :-)

File: cmdstanpy/stanfit.py (hunk 2 above)
Comment (mitzimorris) [reply cmdstanpy/stanfit.py]: thanks - still learning my Python idioms

Comment (mitzimorris) [reply cmdstanpy/stanfit.py]: didn't work - reverting to as it was.

**Pull request merged**